neqo-http3/src/send_message.rs

```diff
@@ -253,25 +173,23 @@ impl SendStream for SendMessage {
     // This method returns if they're still being sent. Request body (if any) is sent by
     // http client afterwards using `send_request_body` after receiving DataWritable event.
     fn has_data_to_send(&self) -> bool {
-        matches!(self.state, SendMessageState::SendingInitialMessage { .. })
+        self.stream.has_buffered_data()
     }
 
     fn close(&mut self, conn: &mut Connection) -> Res<()> {
-        if let SendMessageState::SendingInitialMessage { ref mut fin, .. } = self.state {
-            *fin = true;
-        } else {
-            self.state = SendMessageState::Closed;
-            conn.stream_close_send(self.stream_id)?;
+        self.fin = true;
+        if !self.stream.has_buffered_data() {
+            conn.stream_close_send(self.stream_id())?;
         }
 
         self.conn_events
-            .send_closed(self.stream_id, CloseType::Done);
+            .send_closed(self.stream_id(), CloseType::Done);
         Ok(())
     }
 
     fn stop_sending(&mut self, close_type: CloseType) {
```

Comment:
This function is called when we receive a STOP_SENDING right?  And we're relying on the transport to send the RESET_STREAM in response.  Reading it here in isolation I found the reaction a little confusing.  (The function could be read to be an instruction as in "please stop sending on this stream" as opposed to a reaction "you just got a STOP_SENDING")  A comment on the function might avoid that.  Renaming to handle_stop_sending() might be better.

Reply:
I will rename it to handle_stop_sending and add a comment.